### DISCUSSION TITLE
Reduce Computation time by caclulating SHAP-values only for a subset of features

### DISCUSSION THREAD

**Discussion by GMi-Materials** [shap/shap#2852] [Q&A]
Hello all,
I am using SHAP to compare how different models are interpreting the same feature. We want to check for consistency across different models and confirm their interpretation with literature data.
In my paper we used several different ML models for regression tasks and in one chapter, we want to compare the contribution of each feature of different models. Calculating the SHAP-values for our Gradient Boost and Ridge models is pretty easy, but we want to compare their interpretation also to the ones from ANN, SVM for regression and Gaussian Process Regression models.
I was thinking of using the Deep Explainer and Kernel Explainer for this task. Unfortunately, the computational effort seems to be huge. My dataset contains of ~7000 rows and 440 features.

However, we did identify a small subset of features in our first investigation and it would be nice to calculate SHAP-values for those features only. Can this reduce the computational effort? 
I am aware of the option to use a background dataset, but I specifically want to ask about a reduction in the number of features considered for the calculation of SHAP-values and not not about a reduction of considered samples.

**So long story short: Is there a possibility to calculate SHAP-values via e.g. kernel explainer for only a subset of features and would this reduce the computation time?**
